### PR TITLE
fix CI clang OpenMP not found

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Fix issue that OMP can not be found when using clang++ as compiler.

### CODE REVIEW

**Comment by steindev** [comment]
Now clang++ w/ cuda10 fails...
Error:
```bash
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
 libomp-10-dev : Depends: libomp5-10 (= 1:10.0.0-4ubuntu1~18.04.2) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

File: share/ci/install/clang.sh [hunk 1]
```diff
@@ -11,7 +11,10 @@ clang_version=$(echo $CXX_VERSION | tr -d "clang++-")
 echo "Clang-version: $clang_version"
 
 if ! agc-manager -e clang@${clang_version}; then
-    apt install -y clang-${clang_version}
+    apt install -y clang-${clang_version} libomp-${clang_version}-dev
+    if [[ "$PIC_BACKEND" =~ "omp2b.*" ]] ; then
+        apt install -y libomp-${clang_version}-dev
```
Comment (steindev) [inline share/ci/install/clang.sh:16]: This is the same as the second package two lines above?

Comment (psychocoderHPC) [reply share/ci/install/clang.sh:16]: I forgot to remove libomp from the first install command. I fixed it now. It looks like if I remove the first apt repository in llvm.list all is working fine.

**Comment by SimeonEhrig** [comment]
I spent some time to investigate the problem. The apt sources was changed yesterday:

image

By default, the Ubuntu 20.04 ppa's allows to install clang 6 until 12. If you add the ppa 

```
deb http://apt.llvm.org/focal/ llvm-toolchain-focal main
deb-src http://apt.llvm.org/focal/ llvm-toolchain-focal main
```

no new version is added. For clang 13, 14, 15 you need to add a ppa for each version (the documentation is not complete):

```
# 13
deb http://apt.llvm.org/focal/ llvm-toolchain-focal-13 main
deb-src http://apt.llvm.org/focal/ llvm-toolchain-focal-13 main
# 14
deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main
deb-src http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main
# 15
deb http://apt.llvm.org/focal/ llvm-toolchain-focal-15 main
deb-src http://apt.llvm.org/focal/ llvm-toolchain-focal-15 main
```

**Review by steindev — APPROVED** [review]

**Pull request merged**